R/plotPercentiles.R

```diff
@@ -0,0 +1,107 @@
+#' Comparison plots show user selected variables obtained from different scenario
+#' runs
+
+#' @author Tonn Rueter
+#' @param df `quitte` style data frame containing all variables for each scenario
+#' @param scenarios Character vector contains names of the desired scenarios. If none, all scenarios will be displayed
+#' @param variables Character vector contains names of the desired variables. If none, all variables will be displayed
+#' @importFrom dplyr filter mutate vars
+#' @importFrom reshape2 melt
+#' @importFrom stringr str_extract
+#' @importFrom tidyr pivot_wider
+#' @importFrom ggplot2 ggplot geom_line geom_ribbon facet_wrap facet_grid theme ylab
+#' @export
+plotPercentiles <- function(df, scenarios = NULL, variables = NULL) {
+
+  # In the quitte data frame all perenctiles are given as individual variables
+  # Manipulate input data frame such that all percentiles of a given quantity
+  # are transformed to individual columns. Variable names in the quitte data
+  # frame follow the format "Any|Variable|5.0th Percentile". The regular
+  # expressions below divide the variable name into the prefix and the
+  # percentile specifier
+  data <- df %>%
+    mutate(
+      "percentile" = stringr::str_extract(.data$variable, "[^\\|]+?$"),
+      "variable"   = gsub("\\|[^\\|]+$", "", .data$variable)
+    ) %>%
+    pivot_wider(
+      names_from = "percentile",
+      values_from = "value"
+    )
+
+  # Check which scenarios/variabes are available
+  uniqueScenarios <- unique(data$scenario)
```

Comment:
The following part can be drastically simplified by writing
`if(!all(scenarios %in% unique(data$scenario))) {...}`

Same goes for variables.

Reply:
If you want to list missing variables / scenarios explicitly in a warning, you can use `setdiff(unique(data$scenario), scenarios)` .  For nice printing, you can use 
`paste0(setdiff(unique(data$scenario), scenarios), collapse = ", ")`

Reply:
I don't think we can, since we have three possible outcomes:
1. User does not provide a list of scenarios (scenarios == NULL) -> Use all available scenarios
2. User provides list containing a scenario not present in data -> Throw error
3. User provides list of valid scenarios -> Use those

Helper function `allItemsAvailable` also lets the user know *which* scenario is missing from data

Reply:
Just add a pre-check for outcome 1 (an if-clause), and use setdiff check to determine if outcome 2 or 3 applies otherwise. Also, I added a recommendation for showing warnings above (a combo of `setdiff` and `paste0`).

Reply:
Fixed